python/plugins/db_manager/db_plugins/postgis/connector.py

```diff
@@ -118,29 +117,49 @@ def description(self):
 
         if self._description is None:
 
-            uri = QgsDataSourceUri(self.connection.uri())
+            self._description = []
 
-            # TODO: make this part provider-agnostic
-            sql = self.sql if self.sql.upper().find(' LIMIT ') >= 0 else self.sql + ' LIMIT 1 '
-            uri.setTable('(SELECT row_number() OVER () AS __rid__, * FROM (' + sql + ') as foo)')
-            uri.setKeyColumn('__rid__')
-            # TODO: fetch provider name from connection (QgsAbstractConnectionProvider)
-            # TODO: re-use the VectorLayer for fetching rows in batch mode
-            vl = QgsVectorLayer(uri.uri(False), 'dbmanager_cursor', 'postgres')
+            if re.match('^SHOW', self.sql.strip().upper()):
```

Comment:
Why's this a special case?

Reply:
Because we cannot create a `QgsVector(query)Layer` from that: it's not a SQL expression but an internal command, this will fail:
```sql
(SELECT row_number() OVER () AS __rid__, * FROM (SHOW search_path ) as foo);
```

Reply:
Ok, makes sense. I'm just conscious that this dbmanager specific code has a limited lifetime left, so this statement will break again sometime in the future...

Reply:
I was actually dreaming about a C++ QgsSqlQueryWidget ...

Reply:
> Ok, makes sense. I'm just conscious that this dbmanager specific code has a limited lifetime left, so this statement will break again sometime in the future...

is there a plan to replace DB Manager functionality with something better?

Reply:
Yes: the plan is to integrate DB-manager functionalities into core until we can RIP it forever. 

We already have a growing number of functionalities ported but we aren't quite there yet.

Reply:
[OT here:] Is there a list of the functionalities that will be migrated? Is it worth a QEP?